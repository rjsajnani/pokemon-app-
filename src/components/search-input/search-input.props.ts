export type SearchInputProps = {
  onChangeSearch: (query: string) => void;
  findPokemon: () => void;
  searchText: string;
};
