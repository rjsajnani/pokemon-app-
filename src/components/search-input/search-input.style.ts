import { StyleSheet } from 'react-native';

export const SearchInputStyles = StyleSheet.create({
  searchContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  searchIcon: {
    position: 'absolute',
    right: 10,
  },
  input: {
    flex: 1,
    height: 40,
    paddingHorizontal: 10,
    marginVertical: 12,
    borderRadius: 10,
    borderWidth: 1,
    backgroundColor: '#fff',
    color: '#424242',
  },
});
