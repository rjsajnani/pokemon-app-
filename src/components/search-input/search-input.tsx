import React from 'react';
import { TextInput, View } from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';

import { SearchInputProps as Props } from './search-input.props';
import { SearchInputStyles as styles } from './search-input.style';

export const SearchInput: React.FC<Props> = (props) => {
  const { onChangeSearch, searchText, findPokemon } = props;
  return (
    <View style={styles.searchContainer}>
      <TextInput
        style={styles.input}
        onChangeText={onChangeSearch}
        onEndEditing={() => findPokemon}
        placeholder="Search Pokemon"
        value={searchText}
        clearButtonMode="while-editing"
        onSubmitEditing={findPokemon}
      />
      {searchText.length <= 0 && (
        <Ionicons
          style={styles.searchIcon}
          name={'ios-search'}
          size={15}
          color="#000"
        />
      )}
    </View>
  );
};
