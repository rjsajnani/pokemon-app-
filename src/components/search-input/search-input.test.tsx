import React from 'react';
import { render, fireEvent, RenderAPI } from '@testing-library/react-native';
import { SearchInput } from './search-input';

describe('<SearchInput />', () => {
  const onChangeSearch = jest.fn();
  const findPokemon = jest.fn();
  let instance: RenderAPI;

  beforeEach(function () {
    instance = render(
      <SearchInput
        onChangeSearch={onChangeSearch}
        searchText=""
        findPokemon={findPokemon}
      />,
    );
  });

  it('renders correctly', function () {
    const tree = instance.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly and call onChangeSearch when typing', async () => {
    const { findByPlaceholderText } = instance;

    const inputField = await findByPlaceholderText('Search Pokemon');

    fireEvent.changeText(inputField, 'Pikachu');

    expect(onChangeSearch).toHaveBeenCalledWith('Pikachu');
  });

  it('should call findPokemon when submitting the search', () => {
    const { getByPlaceholderText } = instance;
    const inputField = getByPlaceholderText('Search Pokemon');

    fireEvent.changeText(inputField, 'Charizard');

    fireEvent(inputField, 'onSubmitEditing');

    expect(findPokemon).toHaveBeenCalled();
  });
});
