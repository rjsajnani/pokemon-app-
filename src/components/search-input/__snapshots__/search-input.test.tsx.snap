// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <TextInput
    clearButtonMode="while-editing"
    onChangeText={[MockFunction]}
    onEndEditing={[Function]}
    onSubmitEditing={[MockFunction]}
    placeholder="Search Pokemon"
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "color": "#424242",
        "flex": 1,
        "height": 40,
        "marginVertical": 12,
        "paddingHorizontal": 10,
      }
    }
    value=""
  />
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#000",
          "fontSize": 15,
        },
        {
          "position": "absolute",
          "right": 10,
        },
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ï”…
  </Text>
</View>
`;
